src/ClientRuntime/Microsoft.Rest.ClientRuntime/ServiceClient.cs

```diff
@@ -253,7 +269,7 @@ protected static HttpClientHandler CreateRootHandler()
         /// <summary>
         /// Gets the HttpClient used for making HTTP requests.
         /// </summary>
-        public HttpClient HttpClient { get; protected set; }
+        public virtual HttpClient HttpClient { get; protected set; }
 
```

Comment:
For this one - I still consider it as a potential bug hole. We use Get and Set of the HttpClient property during ServiceClient constructor code. So we put all the responsibility on the developers -> they need to be careful and do not use any objects that are not yet initialized in the overwritten property. 

1. if you walk over the call stack of ServiceClient ctr then HttpClient.Set method is called before any HttpClient.Get method call. 
2. By overriding just Get you will lose whatever was Set in the base class constructor. 
3. In your example in FabricamServiceClient class you completely ignore the HttpClient that was passed in the public FabricamServiceClient(HttpClient httpClient)  constructor (because all your logic of initialization of HttpClient is happening in the HttpClient Get. Then why do you even expose a constructor that takes httpClient parameter?